packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -429,6 +429,24 @@ export class AuthUtil {
 
         return state
     }
+
+    /**
+     * Returns a connection that will work with Amazon Q from the given connections.
+     *
+     * TODO: This method seems like it can be generalized inside the Q AuthUtil. We should eventually
+     * look to add this in to there if possible so that Auth operations are centralized.
+     */
+    findUsableQConnection(connections: AwsConnection[]): AwsConnection | undefined {
```

Comment:
This was moved from `backend_amazonq.ts` since we needed it to be shareable outside of just the UI